### PR TITLE
Add use magic items list picker

### PR DESCRIPTION
**Pull request by ajrb**
Using standard list picker window rather than implementing the small list from the old banner UI mode. I think this is better, and less work since it's reusing the existing component. We can revisit if there are many who want the UI as classic for this.

### CODE REVIEW

File: Assets/Scripts/Game/UserInterfaceWindows/DaggerfallGuildServicePopupWindow.cs [hunk 1]
```diff
@@ -231,7 +231,7 @@ ItemCollection GetMerchantMagicItems()
         ItemCollection GetMerchantPotions()
         {
             ItemCollection potions = new ItemCollection();
-            for (int n = UnityEngine.Random.Range(12, 20); n > 0; n--)
+            for (int n = UnityEngine.Random.Range(16, 24); n > 0; n--)
                 potions.AddItem(ItemBuilder.CreateRandomPotion());
```
[the remark this answers is not in the document]
Comment (Allofich) [reply Assets/Scripts/Game/UserInterfaceWindows/DaggerfallGuildServicePopupWindow.cs]: The close and re-open thing applies for any random selection of things, though, like other shops, right? Wouldn't it be the same as someone who wanted arrows who kept reloading their game and re-entering a shop if they insisted on getting arrows? I guess that just depends how you think about playing the game. I like to play working within the constraints of what the game gave me on the first roll, but there are probably people who reload, too. 

As for building quality, it's also used for the number of magic items sold at a mages guild (like classic). It seems like an OK thing to use to me, to apply to not just shops but also guilds.

Anyway, as I think you know, my feeling is that we shouldn't change values from classic unless they're clearly mistaken, or don't seem to be what was intended/the Daggerfall Chronicles describes something better or more consistent, or the classic values are just clearly not producing satisfactory results, like super-fast running skill advancement or super-slow language skill advancement. I've never used potions much, though, so if you think the classic potions are clearly too few, we could do that. We can always adjust things later if necessary. I'd say use the classic value, though, unless you're confident we shouldn't. Not a huge deal either way.

[the remark this answers is not in the document]
Comment (ajrb) [reply Assets/Scripts/Game/UserInterfaceWindows/DaggerfallGuildServicePopupWindow.cs]: Shop shelves persist their contents for 24 hours after you look, where as potions stock are generated each time - so you don't even need to reload. How about I use the classic for number of potions, and then have a small random stack size for each? 1-4 maybe?

[the remark this answers is not in the document]
Comment (ajrb) [reply Assets/Scripts/Game/UserInterfaceWindows/DaggerfallGuildServicePopupWindow.cs]: Yes classic does that and so far i've not decided how it should change. Also need to find out how classic generates magic items for sale. My testing of classic the items persist  and are deterministic for a given character, but also different for each MG.

File: Assets/Scripts/Game/MagicAndEffects/EntityEffectManager.cs [hunk 2]
```diff
@@ -785,8 +786,14 @@ public void UseItem(DaggerfallUnityItem item)
                         EntityEffectBundle bundle = new EntityEffectBundle(bundleSettings, entityBehaviour);
                         SetReadySpell(bundle, true);
 
-                        // TODO: Apply durability loss to used item on use
+                        // Apply durability loss to used item on use
                         // http://en.uesp.net/wiki/Daggerfall:Magical_Items#Durability_of_Magical_Items
+                        item.currentCondition -= 10;
+                        if (item.currentCondition <= 0 && collection != null)
```
Comment (Interkarma) [inline Assets/Scripts/Game/MagicAndEffects/EntityEffectManager.cs]: Bonus, thank you! :)

Comment (ajrb) [reply Assets/Scripts/Game/MagicAndEffects/EntityEffectManager.cs]: Figured while I was in the vicinity... :+1:

**Pull request merged**